core/sdk-core/src/main/java/software/amazon/awssdk/core/signer/AsyncRequestBodySigner.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright 2010-2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.core.signer;
+
+import software.amazon.awssdk.annotations.SdkPublicApi;
+import software.amazon.awssdk.core.interceptor.ExecutionAttributes;
+import software.amazon.awssdk.http.SdkHttpFullRequest;
+import software.amazon.awssdk.http.async.SdkHttpContentPublisher;
+
+/**
+ * Interface for the signer used for signing the async requests.
+ */
+@SdkPublicApi
+@FunctionalInterface
+public interface AsyncRequestBodySigner {
+    /**
+     * Method that takes in an signed request and async request body provider,
+     * and returns a transformed version the request body provider.
+     *
+     * @param request             The signed request (with Authentication header)
+     * @param executionAttributes Contains the attributes required for signing the request
+     * @return The transformed requestPublisher (with singing operator)
+     */
+    SdkHttpContentPublisher signAsyncRequestBody(SdkHttpFullRequest request, SdkHttpContentPublisher requestPublisher,
```

Comment:
Should we use AsyncRequestBody here instead of SdkHttpContentPublisher? They are kinda the same thing and since this is more public I'd err towards AsyncRequestBody. We'd have to do a bit of refactoring to pass AsyncRequestBody further down the stack and transform only at the MakeAsyncHttpRequest stage.

Reply:
Do you mean we should return AsyncRequestBody in this method and use that method somewhere [here](https://github.com/aws/aws-sdk-java-v2/blob/master/core/sdk-core/src/main/java/software/amazon/awssdk/core/internal/http/pipeline/stages/MakeAsyncHttpRequestStage.java#L98-L100)?